src/antidote.erl

```diff
@@ -126,33 +126,59 @@ read_objects(Objects, TxId) ->
 -spec update_objects([{bound_object(), op_name(), op_param()}], txid())
                     -> ok | {error, reason()}.
 update_objects(Updates, TxId) ->
-    %% Execute each update as in an interactive transaction
-    Results = lists:map(
-                fun({{Key, Type, Bucket}, Op, OpParam}) ->
-                        case clocksi_iupdate(TxId, {Key, Bucket}, Type,
-                                             {Op, OpParam}) of
-                            ok -> ok;
-                            {error, Reason} ->
-                                lager:debug("Update failed. Reason : ~p",[Reason]),
-                                error
-                        end
-                end, Updates),
-    case lists:member(error, Results) of
-       true -> {error, update_failed}; %% TODO: Capture the reason for error
-       false -> ok
+    {_, _, CoordFsmPid} = TxId,
+    Operations = lists:map(fun(Update) ->
+                                {Key, Type, Bucket, Op} = case Update of
+                                    %% the following checks the format of the
+                                    %% operation for compatibility with
+                                    %% some systests that call updates with
+                                    %% {Operation, Params} (as a single parameter),
+                                    %% and Operation, Params (two separate parameters).
+                                    {{K, T, B}, O} ->
+                                        {K, T, B, O};
+                                    {{K, T, B}, O, P} ->
+                                        {K, T, B, {O, P}}
+                                end,
+                                case materializer:check_operations([{update, {{Key, Bucket}, Type, Op}}]) of
+                                    ok ->
+                                        {{Key, Bucket}, Type, Op};
+                                    {error, _Reason} ->
+                                        {error, type_check}
+                                end
+                            end, Updates),
+    case lists:member({error, type_check}, Operations) of
+        true ->
+            {error, type_check};
+        false ->
+            %%            lager:info("gonna start multiple updates: ~p", [Operations]),
+            case gen_fsm:sync_send_event(CoordFsmPid, {update_objects, Operations}, ?OP_TIMEOUT) of
+                ok ->
+                    ok;
+                {aborted, TxId} ->
```

Comment:
yes. I will fix that. Strangely, dialyzer does not complain...
Maybe it can't detect the case where this would happen, as it involves a complex exchange of messages...